Functions/Admin/Backup-RsEncryptionKey.ps1

```diff
@@ -4,71 +4,141 @@
+        
+        .EXAMPLE
+            Backup-RSEncryptionKey -ComputerName "sql2012a243" -Credential (Get-Credential) -ReportServerInstance 'SQL2012' -ReportServerVersion 'SQLServer2012' -Password 'Enter Your Password' -KeyPath 'C:\ReportingServices\Default.snk'
+            Description
+            -----------
+            This command will back up the encryption key against named instance (SQL2012) from SQL Server 2012 Reporting Services.
+            To do so, it will not use the default connection, but rather connect to the computer "sql2012a243", prompting the user for connection credentials.
+        
+        .NOTES
+            Author:      ???
+            Editors:     Friedrich Weinmann
+            Created on:  ???
+            Last Change: 31.01.2017
+            Version:     1.1
+            
+            Release 1.1 (26.01.2017, Friedrich Weinmann)
+            - Fixed Parameter help (Don't poison the name with "(optional)", breaks Get-Help)
+            - Implemented ShouldProcess (-WhatIf, -Confirm)
+            - Replaced calling exit with throwing a terminating error (exit is a bit of an overkill when failing a simple execution)
+            - Improved error message on failure.
+            - Standardized the parameters governing the Report Server connection for consistent user experience.
+            
+            Release 1.0 (???, ???)
+            - Initial Release
     #>
 
-    [cmdletbinding()]
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
     param
     (
+        [Parameter(Mandatory = $True)]
         [string]
-        $SqlServerInstance='MSSQLSERVER',
-
+        $Password,
+        
+        [Parameter(Mandatory = $True)]
         [string]
-        $SqlServerVersion='13',
-
-        [Parameter(Mandatory=$True)]
+        $KeyPath,
+        
+        [Alias('SqlServerInstance')]
         [string]
-        $Password,
+        $ReportServerInstance,
+        
+        [Alias('SqlServerVersion')]
+        [ReportingServicesTools.SqlServerVersion]
+        $ReportServerVersion,
         
-        [Parameter(Mandatory=$True)]
         [string]
-        $KeyPath    
+        $ComputerName,
+        
+        [System.Management.Automation.PSCredential]
+        $Credential
     )
-
-    $rsWmiObject = New-RsConfigurationSettingObject -SqlServerInstance $SqlServerInstance -SqlServerVersion $SqlServerVersion
-
-    Write-Verbose "Retrieving encryption key..."
-    $encryptionKeyResult = $rsWmiObject.BackupEncryptionKey($Password)
-
-    if ($encryptionKeyResult.HRESULT -eq 0) 
-    {
-        Write-Verbose "Success!"
-    }
-    else 
+    
+    if ($ComputerName) { $tempComputerName = $ComputerName }
+    else { $tempComputerName = ([ReportingServicesTools.ConnectionHost]::ComputerName) }
+    if ($ReportServerInstance) { $tempInstanceName = $ReportServerInstance }
+    else { $tempInstanceName = ([ReportingServicesTools.ConnectionHost]::Instance) }
```

Comment:
Already implemented that after your last comment to that point. Consider all functions that use ShouldProcess covered.